File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.cs
public class Solution {
    public IList<string> StringMatching(string[] words) {
       HashSet<string>patterns=new HashSet<string>();
       for(int i=0;i<words.Length;i++){
        for(int j=0;j<words.Length;j++){
            if(i!=j && words[j].Contains(words[i]))
            patterns.Add(words[i]);
        }
       }
       return new List<string>(patterns);
    }
}

File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var stringMatching = function(words) {
    let result =new Set()
    for(i=0;i<words.length;i++){
        for(j=0;j<words.length;j++){
            if(i!==j && words[j].includes(words[i]))
            result.add(words[i])
        }
    }
    return Array.from(result)
};